### PR TITLE
add method to return tickets by stadium sector

### PR DESCRIPTION
**Pull request by chingiz-mazhitov**
This new feature branch needs to be reviewed

### CODE REVIEW

**Comment by chingiz-mazhitov** [comment]
This new feature branch needs to be reviewed

File: Homework-TicketService.iml
Comment (Alexmansar) [inline Homework-TicketService.iml:1]: add to the gitignore please

File: src/TicketService.java [hunk 1]
```diff
@@ -1,37 +1,74 @@
...
+			emptyTicket.toSaveTicketPrice(49.99);
+
+		} catch (IllegalArgumentException e) {
+			System.out.println(e.getMessage());
+		}
+
+	}
+
+	// generate tickets
+	public static Ticket[] getTickets() {
```
Comment (Alexmansar) [inline src/TicketService.java]: could be private

File: src/TicketService.java [hunk 2]
```diff
@@ -1,37 +1,74 @@
...
+			char sector = stadiumSector[i % stadiumSector.length];
+			tickets[i] = new Ticket("ID" + (i + 1),
+					"Hall-" + (i + 1),
+					"00" + (i + 1), eventTime, isPromo, sector, 5.5);
+		}
+		return tickets;
+	}
+
+	// returns tickets according to their stadiumSector
+	public static List<Ticket> getTicketByStadiumSector(char stadiumSector, Ticket[] initial) {
```
Comment (Alexmansar) [inline src/TicketService.java]: could be private

File: src/TicketService.java [hunk 3]
```diff
@@ -1,37 +1,74 @@
...
+					"002",
+					eventTime1);
+			System.out.println(limitedTicket);
+
+			//saving ticket price
+			emptyTicket.toSaveTicketPrice(49.99);
+
+		} catch (IllegalArgumentException e) {
+			System.out.println(e.getMessage());
+		}
```
Comment (Alexmansar) [inline src/TicketService.java]: just a question - in which case it could be?

**Review by Alexmansar — CHANGES_REQUESTED** [review]

**Comment by Alexmansar** [comment]
@chingiz-mazhitov @Nedzeya Hi, please, also solve the conflicts ))
image

**Comment by Nedzeya** [comment]
Why this new pull request was made withoit weating for mearging my JFB-1 ? I have there   private Map<String, Ticket> tickets = new HashMap<>(); in TicketService Class. And in this case new public static List<Ticket> getTicketByStadiumSector(char stadiumSector, Ticket[] initial)  wouldn't work with it. Generating tickets // generate tickets
	public static Ticket[] getTickets() function wasn't in homework. 
	
	Is it ok ?

**Comment by Alexmansar** [comment]
In general, I do not mind small improvements)
About what was created without your changes - perhaps created at different versions of the main branch (forgot to update)

**Comment by Alexmansar** [comment]
@Nedzeya  @chingiz-mazhitov  I guess when you work on merge conflicts, you'll see your (@Nedzeya ) changes you mentioned earlier ))

File: src/Ticket.java [hunk 4]
```diff
@@ -83,6 +83,11 @@ public void setStadiumSector(char sector) {
         this.stadiumSector = sector;
     }
 
+    // added by me
```
Comment (Nedzeya) [inline src/Ticket.java]: It would be better to explain why it was added (// added by me seems unclear).

File: src/TicketService.java (hunk 3 above)
Comment (Nedzeya) [reply src/TicketService.java]: When incorrect Ticket's arguments would be provided.  I throw new IllegalArgumentException with messages in Ticket class's sets.
P.S. Was it a question to me? :)

File: src/Ticket.java (hunk 4 above)
Comment (Alexmansar) [reply src/Ticket.java]: In general, getters and setters are just standard methods without any description.))

**Comment by Nedzeya** [comment]
> @Nedzeya @chingiz-mazhitov I guess when you work on merge conflicts, you'll see your (@Nedzeya ) changes you mentioned earlier ))

@chingiz-mazhitov please, update your branch JFB-2 from main. (rebase)
You will see changes which block your PR

**Review by Alexmansar — DISMISSED** [review]
@chingiz-mazhitov @Nedzeya could be merged) just resolve conflicts)

**Comment by chingiz-mazhitov** [comment]
> > @Nedzeya @chingiz-mazhitov I guess when you work on merge conflicts, you'll see your (@Nedzeya ) changes you mentioned earlier ))
> 
> @chingiz-mazhitov please, update your branch JFB-2 from main. (rebase) You will see changes which block your PR

I guess now everything is all right and finally ready for merging PR, but I'll prefer to be extra careful here, so if you consider it all right too, you can merge pull request : )

File: src/Ticket.java [hunk 5]
```diff
@@ -70,6 +79,17 @@ public void setEventCode(String eventCode) {
         this.eventCode = eventCode;
     }
 
+    public void setStadiumSector(char sector) {
+        if (sector < 'A' || sector > 'C') {
```
Comment (Nedzeya) [inline src/Ticket.java]: This method is unnecessary here since I have a Sector enum.

File: src/Ticket.java [hunk 6]
```diff
@@ -43,6 +44,10 @@ public Ticket(String concertHall,
         this.time = time;
     }
 
+    public Ticket(String id) {
+        setId(id);
```
Comment (Nedzeya) [inline src/Ticket.java]: This constructor is unnecessary here. It's not used.

File: src/Ticket.java [hunk 7]
```diff
@@ -70,6 +79,17 @@ public void setEventCode(String eventCode) {
...
 
+    public void setStadiumSector(char sector) {
+        if (sector < 'A' || sector > 'C') {
+            throw new IllegalArgumentException("Stadium sector must be between 'A' and 'C'.");
+        }
+        this.stadiumSector = sector;
+    }
+
+    public char getStadiumSector() {
+        return this.stadiumSector;
```
Comment (Nedzeya) [inline src/Ticket.java]: This won't compile.

File: src/TicketService.java (hunk 1 above)
Comment (Nedzeya) [reply src/TicketService.java]: If you want to generate tickets, the method should be inside a try-catch block, and you should use the addTicket function.

File: src/TicketService.java (hunk 2 above)
Comment (Nedzeya) [reply src/TicketService.java]: The arguments here are incorrect

**Comment by Nedzeya** [comment]
> > > @Nedzeya @chingiz-mazhitov I guess when you work on merge conflicts, you'll see your (@Nedzeya ) changes you mentioned earlier ))
> > 
> > 
> > @chingiz-mazhitov please, update your branch JFB-2 from main. (rebase) You will see changes which block your PR
> 
> I guess now everything is all right and finally ready for merging PR, but I'll prefer to be extra careful here, so if you consider it all right too, you can merge pull request : )

Unfortunately, it's not ready for merging because the code doesn't compile. @chingiz-mazhitov, please check the new comments.

**Comment by Nedzeya** [comment]
@Alexmansar, could you please tell me, in real projects, who is supposed to merge a pull request? The person who created it and wrote the code, or the owner of the project?

**Review by Alexmansar — DISMISSED** [review]
Could be merged)

**Review by Alexmansar — DISMISSED** [review]
Could be merged))

**Review by Nedzeya — DISMISSED** [review]

**Review by Nedzeya — DISMISSED** [review]

**Review by Nedzeya — DISMISSED** [review]

**Review by Nedzeya — APPROVED** [review]

**Review by Alexmansar — APPROVED** [review]
Could be merged)

**Comment by Alexmansar** [comment]
> @Alexmansar, could you please tell me, in real projects, who is supposed to merge a pull request? The person who created it and wrote the code, or the owner of the project?

@Nedzeya 
Depending on the project
On my previous one, we used the same approach as here. A developer writes code, creates a PR, asks another colleague to check, if everything is ok the second colleague puts an approve and the first one merges.
Now I have a different approach - the beginning is the same, but at the end the one who did the review and put an approve is the one who merges.
At my friend's project the situation is such that only the team lead can merge. In this case, they have a lot of open PR, and then a lot of conflicts.

**Pull request merged**